operator/pkg/helmreconciler/prune.go

```diff
@@ -193,6 +209,7 @@ func (h *HelmReconciler) GetPrunedResources(revision string, includeClusterResou
 		labels[IstioComponentLabelStr] = componentName
 	}
 	selector := klabels.Set(labels).AsSelectorPreValidated()
+	h.modifyPruneResourcesBasedOnK8SVersion()
```

Comment:
Maybe could be `gvkList := h.GetResourcesByVersion()` instead of mutating globals?

Reply:
+1, think we'd have to change this in a few places though?

Reply:
there are only two places for calling modifyPruneResourcesBasedOnK8SVersion(), one is during install, and the other is during uninstall, so they are used in different context and should be safe. But I can change it to avoid confusion